cmd/account/cli.go

```diff
@@ -71,20 +71,29 @@ func (o *cliOptions) complete(cmd *cobra.Command) error {
 }
 
 func (o *cliOptions) run() error {
-	awsClient, err := o.k8sclusterresourcefactory.GetCloudProvider(o.verbose)
+	_, err := o.k8sclusterresourcefactory.GetCloudProvider(o.verbose)
 	if err != nil {
 		return err
 	}
 
-	credentials, err := awsprovider.GetAssumeRoleCredentials(awsClient, &o.k8sclusterresourcefactory.Awscloudfactory.ConsoleDuration,
-		o.k8sclusterresourcefactory.Awscloudfactory.CallerIdentity.UserId,
-		aws.String(fmt.Sprintf("arn:aws:iam::%s:role/%s",
-			o.k8sclusterresourcefactory.AccountID,
-			o.k8sclusterresourcefactory.Awscloudfactory.RoleName)))
-	if err != nil {
-		return err
+	creds := o.k8sclusterresourcefactory.Awscloudfactory.Credentials
```

Comment:
ah okay I see now, the GetCloudProvider assumes the right role and places the credentials here. I think that GetCloudProvider has too much logic in it then if there are cases where we don't use the awsClient it provides. There can also be cases where we want the root level creds and not assume a role.

Reply:
A fair point, is that necessary for this commit/change? Could that be a future enhancement?